smoke-test/src/test/java/org/opennms/smoketest/minion/MinionHeartbeatOutageTest.java

```diff
@@ -0,0 +1,282 @@
+import org.opennms.netmgt.dao.hibernate.MinionDaoHibernate;
+import org.opennms.netmgt.dao.hibernate.NodeDaoHibernate;
+import org.opennms.netmgt.events.api.EventConstants;
+import org.opennms.netmgt.model.OnmsEvent;
+import org.opennms.netmgt.model.OnmsMonitoringSystem;
+import org.opennms.netmgt.model.OnmsNode;
+import org.opennms.netmgt.model.minion.OnmsMinion;
+import org.opennms.smoketest.NullTestEnvironment;
+import org.opennms.smoketest.OpenNMSSeleniumTestCase;
+import org.opennms.smoketest.utils.DaoUtils;
+import org.opennms.smoketest.utils.HibernateDaoFactory;
+import org.opennms.test.system.api.AbstractTestEnvironment;
+import org.opennms.test.system.api.NewTestEnvironment.ContainerAlias;
+import org.opennms.test.system.api.TestEnvironment;
+import org.opennms.test.system.api.TestEnvironmentBuilder;
+import org.opennms.test.system.api.utils.SshClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.spotify.docker.client.DockerClient;
+import com.spotify.docker.client.exceptions.DockerException;
+
+/**
+ * This test starts up Minion with the default JMS sink and makes sure
+ * that heartbeat messages continue to be processed even if the Minion
+ * and OpenNMS instances are restarted.
+ * 
+ * @author Seth
+ */
+public class MinionHeartbeatOutageTest {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MinionHeartbeatOutageKafkaTest.class);
+
+    @Rule
+    public TestEnvironment testEnvironment = getTestEnvironment();
+
+    @Rule
+    public Timeout timeout = new Timeout(20, TimeUnit.MINUTES);
+
+    protected HibernateDaoFactory daoFactory;
+
+    public final TestEnvironment getTestEnvironment() {
+        if (!OpenNMSSeleniumTestCase.isDockerEnabled()) {
+            return new NullTestEnvironment();
+        }
+        try {
+            return getEnvironmentBuilder().build();
+        } catch (final Throwable t) {
+            throw new RuntimeException(t);
+        }
+    }
+
+    /**
+     * Override this method to customize the test environment.
+     */
+    protected TestEnvironmentBuilder getEnvironmentBuilder() {
+        final TestEnvironmentBuilder builder = TestEnvironment.builder().all();
+        OpenNMSSeleniumTestCase.configureTestEnvironment(builder);
+        return builder;
+    }
+
+    @Before
+    public void checkForDocker() {
+        Assume.assumeTrue(OpenNMSSeleniumTestCase.isDockerEnabled());
+    }
+
+    @Before
+    public void setup() {
+        // Connect to the postgresql container
+        final InetSocketAddress pgsql = testEnvironment.getServiceAddress(ContainerAlias.POSTGRES, 5432);
+        this.daoFactory = new HibernateDaoFactory(pgsql);
+    }
+
+    /**
+     * Install the Kafka features on Minion.
+     * 
+     * @param minionSshAddr
+     * @param kafkaAddress
+     * @throws Exception
+     */
+    protected static void installFeaturesOnMinion(InetSocketAddress minionSshAddr) throws Exception {
```

Comment:
Do we need this? We don't isntall anything any we already print the list of features installed after the Minion is spawned. Same for the OpenNMS instance.